src/bindings/python/tests/test_graph/test_custom_op.py

```diff
@@ -197,3 +199,93 @@ def test_custom_op():
     input_tensor = Tensor(input_data)
     results = request.infer({"data": input_tensor})
     assert np.allclose(results[list(results)[0]], expected_output, 1e-4, 1e-4)
+
+
+class CustomSimpleOp(Op):
+    def __init__(self, inputs=None):
+        super().__init__(self)
+        if inputs is not None:
+            self.set_arguments(inputs)
+            self.constructor_validate_and_infer_types()
```

Comment:
This pattern with first calling `super().__init__(self)` and then `self.set_arguments(inputs)` is repeated everywhere. But in C++, we have `Op(const OutputVector& arguments)` constructor that should give you a way to initialize `Op` with a single `supe().__init__(self, inputs)` even if `inputs` is `None`.

Reply:
Also, `constructor_validate_and_infer_types` is a low-level thing that is easy to forget to call. With `if inputs is not None` thing it still looks bulky and boilerplate. Can we try to call `constructor_validate_and_infer_types` inside `super().__init__` for simple cases? I see obvious difficulties doing that because you cannot clearly recognize the case of a "dummy" initialization in which case `super().__init__` shouldn't call `construct_validate_and_infer_types`, but probably you can see some pythonic ways to do that.

Reply:
So, ideally we should have a minimalistic construct in the form:
```
def __init__(self, inputs=None):
    super().__init__(self, inputs)
```
We should allow `super().__init__` to call `constructor_validate_and_infer_types` depending on `inputs is None` in this approach. So normally when a user creates an op instance he/she provides it with arguments. If the op doesn't really expect any inputs (strange op, like a procedural constant or a node with a side effect that is not really a legal case, so it should be rare), then it should be defined as:
```
def __init__(self, inputs=[]):
    super().__init__(self, inputs)
```
And super init will call `construct_validate_and_infer_types` in this case. Attributes should be set before super init call  if they affect node validation code.

Flimsy?

Reply:
@akuporos 
FYI: If operator is created without inputs (and attributes) by default ctor then in most cases validation will throw. This default created operators are used in GPU plugin but they are not validated.

Reply:
>  If operator is created without inputs

This way of operation instantiating should always be used in IR deserialization. This is what I mentioned as "dummy" initialization. The tricky thing is whether to allow 0 number of inputs or not is up to Op author and cannot be firgured out somehow inside super init without additional hints.